[Fleet] Fix upgrades for packages with restructured inputs

kpollich opened the pull request:
## Summary

Addresses errors surfaced when testing upgrades from AWS 0.6.1 to 0.10.4. Namely, when inputs are removed from a package between versions we were initially throwing errors for each input in the new package that didn't exist on the outdated package version. Now, we instead simply skip over cases like this in which an input no longer exists on the new package version.

## Screen Recording

![Kapture 2021-08-24 at 13 03 32](https://user-images.githubusercontent.com/6766512/130659417-b6b537af-7fb2-4636-98a6-3b7b43699ee1.gif)

I've also included a validation fix for when streams are removed that addresses issues with the Nginx package. 

Fixes #109808 
Fixes #109805

elasticmachine commented:
Pinging @elastic/fleet (Team:Fleet)

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/148264/)
* Commit: 80246de3bf1c8e1a643797c36b129e68aa5f6300
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-109887/80246de3bf1c8e1a643797c36b129e68aa5f6300)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @kpollich

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/148264/","number":"148264","commit":"80246de3bf1c8e1a643797c36b129e68aa5f6300"}]}
PIPELINE-->

jen-huang reviewed (approved):
Any test coverage we can add for this case?

kpollich commented:
> Any test coverage we can add for this case?

Good call. We should be able to add a test package version that incurs this case. I'll work on that this morning.

kpollich commented:
@jen-huang I added some basic test cases for input and policy template restructures in https://github.com/elastic/kibana/pull/109887/commits/5889928fe426f2744fb0881feb70bd0fb87c21dd. 

I think it'd be good to add some more robust assertions to these tests at some point rather than just asserting on success and error values. e.g. actually verifying the expected data in the updated policy objects. I added a tech debt issue to the backlog to capture that improvement: https://github.com/elastic/kibana/issues/110087

amolnater-qasource commented:
Hi @EricDavisX 
We are validating this PR in lieu of reported issue at https://github.com/elastic/kibana/issues/109805
We have revalidated this on 7.15.0 BC-3 and found it **fixed** now.

**Observations:**
- We are now able to upgrade Nginx `v0.2.4` to `v0.7.0` on 7.15.0 BC Kibana.

**Build details:**
Build 43818
Commit: 6f7562b1906dcfad65809da8fdec15df353d0252

**Screen Recording:**

https://user-images.githubusercontent.com/77374876/131457724-7f190c84-66c7-4e35-a672-51ebc6df327a.mp4

Thanks
